spec/unit/matrix-client.spec.ts

```diff
@@ -165,12 +205,28 @@ describe("MatrixClient", function() {
             }
             return Promise.resolve(next.data);
         }
-        // Jest doesn't let us have custom expectation errors, so if you're seeing this then
-        // you forgot to handle at least 1 pending request. Check your tests to ensure your
-        // number of expectations lines up with your number of requests made, and that those
-        // requests match your expectations.
-        expect(true).toBe(false);
-        return new Promise(() => {});
+
+        const receivedRequestQueryString = new URLSearchParams(
+            convertQueryDictToStringRecord(queryParams),
+        ).toString();
+        const receivedRequest = decorateStringWithAnsiColor(
+            `${method} ${prefix}${path}${receivedRequestQueryString}`,
+            AnsiColorCode.Red,
+        );
+        const expectedQueryString = new URLSearchParams(
+            convertQueryDictToStringRecord(next.expectQueryParams),
+        ).toString();
+        const expectedRequest = decorateStringWithAnsiColor(
+            `${next.method} ${next.prefix ?? ''}${next.path}${expectedQueryString}`,
+            AnsiColorCode.Green,
+        );
+        // If you're seeing this then you forgot to handle at least 1 pending request.
+        throw new Error(
+            `A pending request was not handled: ${receivedRequest} ` +
+            `(next request expected was ${expectedRequest})\n` +
+            `Check your tests to ensure your number of expectations lines up with your number of requests ` +
+            `made, and that those requests match your expectations.`,
+        );
```

Comment:
Better error message:

<img width="1508" alt="Colored terminal output of what this assertion looks like" src="https://user-images.githubusercontent.com/558581/204938798-f745e4be-6952-4bff-8e78-bfd03490e781.png">